### PR TITLE
6876 refactored boxesPlatformVersions

### PR DESCRIPTION
**Pull request by kilfat**
Diveded boxesPlatformVersions function to showBoxesPlatformVersions and getBoxesPlatformVersions.

### CODE REVIEW

File: core/session.rb [hunk 1]
```diff
@@ -763,46 +763,45 @@ def showProvider(name)
...
+    boxesManager.each do |box, params|
+      next if params['platform'] != boxPlatform # skip unknown platform
+      if params.has_value?(boxPlatform)
         box_platform_version = params['platform_version']
         boxes_versions.push(box_platform_version)
         exit_code = 0
       else
-        $out.error "#{$session.boxPlatform} has 0 supported versions! Please check box platform!"
-        exit_code = 1
+        raise "#{boxPlatform} has 0 supported versions! Please check box platform!"
```
Comment (zmm) [inline core/session.rb:796]: Better and more compact way to describe exceptions is a following:

if error 
   raise "Error occured"

File: core/session.rb [hunk 2]
```diff
@@ -763,46 +763,45 @@ def showProvider(name)
...
     # get boxes platform versions
-    $session.boxes.boxesManager.each do |box, params|
-      next if params['platform'] != $session.boxPlatform # skip unknown platform
-      if params.has_value?($session.boxPlatform)
+    boxesManager.each do |box, params|
+      next if params['platform'] != boxPlatform # skip unknown platform
+      if params.has_value?(boxPlatform)
         box_platform_version = params['platform_version']
         boxes_versions.push(box_platform_version)
         exit_code = 0
```
Comment (galiaf95) [inline core/session.rb]: This line is unnecessary

File: core/session.rb [hunk 3]
```diff
@@ -763,46 +763,44 @@ def showProvider(name)
...
+      raise "Specify parameter --platforms and try again"
     end
 
     # check for supported platforms
     some_platform = $session.boxes.boxesManager.find { |box| box[1]['platform'] == $session.boxPlatform }
     if some_platform.nil?
-      $out.error "Platform #{$session.boxPlatform} is not supported!"
-      exit_code = 1
+      raise  "Platform #{$session.boxPlatform} is not supported!"
     else
```
Comment (zmm) [inline core/session.rb:776]: This else is not needed - instead you can just write 

```
if some_platform.nil?
 raise  "Platform #{$session.boxPlatform} is not supported!"
end
$out.info "Supported versions for #{$session.boxPlatform}:"
```

File: core/session.rb [hunk 4]
```diff
@@ -763,46 +763,44 @@ def showProvider(name)
...
+    boxesManager.each do |box, params|
+      next if params['platform'] != boxPlatform # skip unknown platform
+      if params.has_value?(boxPlatform)
         box_platform_version = params['platform_version']
         boxes_versions.push(box_platform_version)
-        exit_code = 0
       else
-        $out.error "#{$session.boxPlatform} has 0 supported versions! Please check box platform!"
-        exit_code = 1
+        raise "#{boxPlatform} has 0 supported versions! Please check box platform!"
```
Comment (zmm) [inline core/session.rb:796]: Old comment (keep in mind the comment above about else):

> Better and more compact way to describe exceptions is a following
> if error 
> raise "Error occured"

**Pull request merged**